File: app/src/main/java/org/stepic/droid/view/dialogs/LoadingProgressDialog.java
package org.stepic.droid.view.dialogs;

import android.app.ProgressDialog;
import android.content.Context;
import android.support.annotation.StringRes;

import org.stepic.droid.R;

public class LoadingProgressDialog extends ProgressDialog {
    public LoadingProgressDialog(Context context, @StringRes int titleRes) {
        super(context);
        setTitle(context.getString(titleRes));
        setMessage(context.getString(R.string.loading_message));
        setCancelable(false);
    }

    public LoadingProgressDialog(Context context) {
        this(context, R.string.loading);
    }

}

File: app/src/main/java/org/stepic/droid/ui/dialogs/LoadingProgressDialog.kt
package org.stepic.droid.ui.dialogs

import android.app.Dialog
import android.content.Context
import android.os.Bundle
import android.view.Window
import androidx.annotation.StringRes
import kotlinx.android.synthetic.main.dialog_progress.*
import org.stepic.droid.R

class LoadingProgressDialog
@JvmOverloads
constructor(
    context: Context,
    @StringRes private val titleRes: Int = R.string.loading
) : Dialog(context) {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        requestWindowFeature(Window.FEATURE_NO_TITLE)
        setContentView(R.layout.dialog_progress)
        progressDialogTitle.setText(titleRes)
        progressDialogMessage.setText(R.string.loading_message)
        setCanceledOnTouchOutside(false)
        setCancelable(false)
    }
}